uitools/import/Esri/ArcGISRuntime/Toolkit/Callout.qml

```diff
@@ -404,692 +333,349 @@ Item {
-        visible: false
-        z: 100
-        clip: true
-        color: "transparent"
-
-        Canvas {
-            id: canvas
-            anchors.fill: parent
-
-            property bool createPathAndPaint: false
+    Component.onCompleted: {
+        background.children.push(shapeTail.createObject())
+    }
 
-            antialiasing: true
+    contentItem: GridLayout {
+        id: calloutLayout
+        columns: 3
+        rows: 2
+        columnSpacing: 7
+        Loader {
+            sourceComponent: calloutContent
             clip: true
-
-            // work around for Qt bug with Canvas on iOS.
-            // Rendering to Frame buffer object causes weirdness with size.
-            Component.onCompleted: {
-                if (Qt.platform.os === "ios")
-                    renderTarget = Canvas.Image;
+            Layout.columnSpan: 3
+            Layout.rowSpan: 2
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+            Layout.preferredWidth: root.autoAdjustWidth ? root.maxWidth : root.calloutWidth
+            visible: calloutContent
+        }
+        Image {
+            id: image
+            source: calloutData ? calloutData.imageUrl : ""
+            Layout.rowSpan: 2
+            Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
+            Layout.fillHeight: true
+            Layout.preferredWidth: 40
+            fillMode : Image.PreserveAspectFit
+            visible: !calloutContent && source && source.toString() !== ""
+        }
+        Label {
+            id: title
+            text: calloutData ? calloutData.title : ""
+            wrapMode: Text.Wrap
+            clip: true
+            elide: Text.ElideRight
+            // Is visible (even when empty) if detail is visible, otherise
+            // row & columnspan offsets go askew.
+            visible: !calloutContent && (text || detail.visible)
+            Layout.alignment: Qt.AlignVCenter
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+            Layout.preferredWidth: autoAdjustWidth ? -1 : internal.labelWidthFrom.bind(this)(root.calloutWidth)
+            Layout.maximumWidth: autoAdjustWidth ? internal.labelWidthFrom.bind(this)(root.maxWidth) : -1
+            Layout.columnSpan: {
+                let span = 1;
+                if (!accessoryButton.visible)
+                    span++;
+                if (!image.visible)
+                    span++;
+                return span;
             }
-
-            renderTarget: Canvas.FramebufferObject
-            renderStrategy: Canvas.Cooperative
-
-            // handler to override for drawing
-            onPaint: {
-                screenCoordinates.x = borderWidth;
-                screenCoordinates.y = borderWidth;
-                drawCalloutFrame();
+            Layout.rowSpan: {
+                let span = 1;
+                if (!detail.visible)
+                    span++;
+                return span;
```

Comment:
🙃 
```suggestion
                    span++;

                return span;
```

There are a few of these. I won't comment on them all but you get the idea. Optional throughout.